main.py

```diff
@@ -670,6 +670,31 @@ def delete_link_metadata(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         return JSONResponse("Operation successful")
 
+    @rest('v3/links/{link_id}', methods=['DELETE'])
+    def delete_link(self, request: Request) -> JSONResponse:
+        """Delete a down link from topology. It won't work for up links."""
+        link_id = request.path_params["link_id"]
+        try:
+            link = self.links[link_id]
+        except KeyError:
+            raise HTTPException(404, detail="Link not found.")
+        if link.status != EntityStatus.DOWN:
+            raise HTTPException(409, detail="Link is not down.")
+        if link.endpoint_a.link and link == link.endpoint_a.link:
+            switch = link.endpoint_a.switch
+            link.endpoint_a.link = None
+            link.endpoint_a.nni = False
+            self.topo_controller.upsert_switch(switch.id, switch.as_dict())
+        if link.endpoint_a.link and link == link.endpoint_b.link:
+            switch = link.endpoint_b.switch
+            link.endpoint_b.link = None
+            link.endpoint_b.nni = False
+            self.topo_controller.upsert_switch(switch.id, switch.as_dict())
+        link = self.links.pop(link_id)
+        self.topo_controller.delete_link(link_id)
+        self.notify_topology_update()
```

Comment:
Notifying that the topology has been updated is enough for `pathfinder` to update its data. `coloring` currently can't delete any flows and there are leftover flows related to the delete link.

Reply:
Yes, we'll need `self.notify_topology_update()` here. Good catch, for `coloring` check out the other thread that I opened, let me know if you have any other better idea or suggestion.

Reply:
Following [here](https://github.com/kytos-ng/topology/pull/179#discussion_r1418022149).